Source/missiles.cpp

```diff
@@ -387,130 +387,77 @@ void RotateBlockedMissile(Missile &missile)
 	int my = position.y;
 
+	bool isMonsterHit = false;
+	const int mid = dMonster[mx][my];
 	if (missile._micaster != TARGET_BOTH && !missile.IsTrap()) {
 		if (missile._micaster == TARGET_MONSTERS) {
-			int mid = dMonster[mx][my];
 			if (mid != 0 && (mid > 0 || Monsters[abs(mid) - 1]._mmode == MonsterMode::Petrified)) {
-				if (MonsterMHit(
-				        missile._misource,
-				        abs(mid) - 1,
-				        mindam,
-				        maxdam,
-				        missile._midist,
-				        missile._mitype,
-				        shift)) {
-					if (!nodel)
-						missile._mirange = 0;
-					missile._miHitFlag = true;
-				}
-			}
-			if (dPlayer[mx][my] > 0
-			    && dPlayer[mx][my] - 1 != missile._misource
-			    && Plr2PlrMHit(
-			        missile._misource,
-			        dPlayer[mx][my] - 1,
-			        mindam,
-			        maxdam,
-			        missile._midist,
-			        missile._mitype,
-			        shift,
-			        &blocked)) {
-				if (gbIsHellfire && blocked) {
-					RotateBlockedMissile(missile);
-				} else if (!nodel) {
-					missile._mirange = 0;
-				}
-				missile._miHitFlag = true;
+				isMonsterHit = MonsterMHit(missile._misource, abs(mid) - 1, minDam, maxDam, missile._midist, missile._mitype, shift);
 			}
 		} else {
 			auto &monster = Monsters[missile._misource];
 			if ((monster._mFlags & MFLAG_TARGETS_MONSTER) != 0
-			    && dMonster[mx][my] > 0
-			    && (Monsters[dMonster[mx][my] - 1]._mFlags & MFLAG_GOLEM) != 0
-			    && MonsterTrapHit(dMonster[mx][my] - 1, mindam, maxdam, missile._midist, missile._mitype, shift)) {
-				if (!nodel)
-					missile._mirange = 0;
-				missile._miHitFlag = true;
-			}
-			if (dPlayer[mx][my] > 0
-			    && PlayerMHit(
-			        dPlayer[mx][my] - 1,
-			        &monster,
-			        missile._midist,
-			        mindam,
-			        maxdam,
-			        missile._mitype,
-			        shift,
-			        0,
-			        &blocked)) {
-				if (gbIsHellfire && blocked) {
-					RotateBlockedMissile(missile);
-				} else if (!nodel) {
-					missile._mirange = 0;
-				}
-				missile._miHitFlag = true;
-			}
+			    && mid > 0
+			    && (Monsters[mid - 1]._mFlags & MFLAG_GOLEM) != 0)
+				isMonsterHit = MonsterTrapHit(mid - 1, minDam, maxDam, missile._midist, missile._mitype, shift);
 		}
 	} else {
-		int mid = dMonster[mx][my];
 		if (mid > 0) {
-			mid -= 1;
-			if (missile._micaster == TARGET_BOTH) {
-				if (MonsterMHit(
-				        missile._misource,
-				        mid,
-				        mindam,
-				        maxdam,
-				        missile._midist,
-				        missile._mitype,
-				        shift)) {
-					if (!nodel)
-						missile._mirange = 0;
-					missile._miHitFlag = true;
-				}
-			} else if (MonsterTrapHit(mid, mindam, maxdam, missile._midist, missile._mitype, shift)) {
-				if (!nodel)
-					missile._mirange = 0;
-				missile._miHitFlag = true;
-			}
+			if (missile._micaster == TARGET_BOTH)
+				MonsterMHit(missile._misource, mid - 1, minDam, maxDam, missile._midist, missile._mitype, shift);

```

Comment:
```suggestion
				isMonsterHit = MonsterMHit(missile._misource, mid - 1, minDam, maxDam, missile._midist, missile._mitype, shift);
```

Reply:
:sweat_smile:  oops

Reply:
after review I recommend squash, to not have not working commit in history.